components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -1,42 +1,78 @@
-import { Component, Input, Output, EventEmitter } from '@angular/core';
-import { Desktop } from 'app/entities/desktop/desktop.model';
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Desktop, DailyNodeRunsStatus } from 'app/entities/desktop/desktop.model';
 import { DateTime } from 'app/helpers/datetime/datetime';
+import { Store } from '@ngrx/store';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { GetDailyNodeRunsStatusTimeSeries } from 'app/entities/desktop/desktop.actions';
+import { getDailyNodeRuns } from 'app/entities/desktop/desktop.selectors';
 
-const historyTypes = [
-  { text: 'converged', icon: 'check_box' },
-  { text: 'unchanged', icon: 'indeterminate_check_box' },
-  { text: 'error', icon: 'error' },
-  { text: 'unknown', icon: 'help' }
-];
-
-const checkinHistory = Array.from(new Array(29)).map((_v, i) => {
-  const historyType = historyTypes[Math.floor(Math.random() * historyTypes.length)];
-  const date = new Date();
-  return { ...historyType, date: new Date(date.setDate(date.getDate() - i)) };
-});
+// const checkinHistory = Array.from(new Array(29)).map((_v, i) => {
+//   const historyType = historyTypes[Math.floor(Math.random() * historyTypes.length)];
+//   const date = new Date();
+//   return { ...historyType, date: new Date(date.setDate(date.getDate() - i)) };
+// });
 
 @Component({
   selector: 'app-desktop-detail',
   templateUrl: './desktop-detail.component.html',
   styleUrls: ['./desktop-detail.component.scss']
 })
-export class DesktopDetailComponent {
+export class DesktopDetailComponent implements OnInit {
 
   @Input() desktop: Desktop;
   @Input() fullscreened = false;
 
   @Output() closed: EventEmitter<any> = new EventEmitter();
   @Output() fullscreenToggled: EventEmitter<void> = new EventEmitter();
 
+  public checkInHistory: DailyNodeRunsStatus[];
   public ceil = Math.ceil;
   public DateTime = DateTime;
   public showCheckinDebug = false;
   public checkinTableType = 'grid';
   public checkinGridFlexType = 'wrap';
   public checkinNumDays = 14;
+  public historyIcons = {
+    converged: 'check_box',
+    unchanged: 'indeterminate_check_box',
+    failure: 'warning',
+    error: 'warning',
+    unknown: 'help',
+    missing: 'help'
+  };
+
+  constructor(
+    private store: Store<NgrxStateAtom>
+  ) {
+    this.store.select(getDailyNodeRuns).subscribe((dailyNodeRuns) => {
```

Comment:
Need to add lifecycle management (`OnDestroy` and `isDestroyed` boilerplate) on the subscription.